exec/java-exec/src/main/java/org/apache/drill/exec/memory/AtomicRemainder.java

```diff
@@ -76,16 +74,17 @@ public void setLimit(long limit) {
 
   }
   /**
-   * Automatically allocate memory. This is used when an actual allocation happened to be larger than requested. This
-   * memory has already been used up so it must be accurately accounted for in future allocations.
+   * Automatically allocate memory. This is used when an actual allocation happened to be larger than requested, or when
+   * a buffer has it's ownership passed to another allocator.<br>
+   * This memory has already been used up so it must be accurately accounted for in future allocations.
    *
-   * @param size
+   * @param size extra allocated memory that needs to be accounted for
    */
   public boolean forceGet(long size) {
     if (get(size, this.applyFragmentLimit)) {
       return true;
     } else {
-      availableShared.addAndGet(size);
+      availableShared.addAndGet(-size);
```

Comment:
It can become negative if the allocator takes ownership of a buffer and exceeds it's maximum allocated memory. Negative values are handled properly, at least that's my understanding